app/src/main/java/com/example/linkdevmvvm/ui/base/BaseActivityForDrawer.java

```diff
@@ -0,0 +1,133 @@
+package com.example.linkdevmvvm.ui.base;
+
+import android.support.design.widget.NavigationView;
+import android.support.v4.widget.DrawerLayout;
+import android.support.v7.app.ActionBarDrawerToggle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.view.Menu;
+import android.widget.FrameLayout;
+
+import com.example.linkdevmvvm.R;
+import com.example.linkdevmvvm.adapters.CustomDrawerAdapter;
+import com.example.linkdevmvvm.dataModel.DrawerItem;
+import com.example.linkdevmvvm.utils.Utils;
+
+import java.util.ArrayList;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public abstract class BaseActivityForDrawer extends BaseActivity implements CustomDrawerAdapter.OnItemSideMenuClicked {
+
+    @BindView(R.id.flContent)
+    public FrameLayout flContent;
+    @BindView(R.id.load_view)
+    public FrameLayout loadView;
+    @BindView(R.id.toolbar)
+    public Toolbar toolbar;
+    @BindView(R.id.rv_menuList)
+    RecyclerView rvMenuList;
+    @BindView(R.id.nvView)
+    NavigationView nvView;
+    @BindView(R.id.drawer_layout)
+    DrawerLayout drawerLayout;
+    private CustomDrawerAdapter customDrawerAdapter;
+
+    @Override
+    public void setContentView(int layoutResID) {
+        DrawerLayout fullView = (DrawerLayout) getLayoutInflater().inflate(R.layout.activity_base_for_drawer, null);
+        FrameLayout activityContainer = fullView.findViewById(R.id.flContent);
+        getLayoutInflater().inflate(layoutResID, activityContainer, true);
+        super.setContentView(fullView);
+
+
+    }
+
+    @Override
```

Comment:
The inflation of the options menu should not be the BaseActivityForDrawer instead it should be specific to each activity that extends the BaseActivityForDrawer